File: Two Pointers/Is Subsequence/isSubsequence.rs
impl Solution {
    pub fn is_subsequence(s: String, t: String) -> bool {
        let mut t_chars: Vec<char> = t.chars().collect();
        for c in s.chars() {
            if let Some(i) = t_chars.iter().position(|&x| x == c) {
                t_chars = t_chars[(i + 1)..].to_vec();
            } else {
                return false;
            }
        }
        true
    }
}

File: Two Pointers/Is Subsequence/isSubsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int j = 0;

        if (s.length() < 1) {
            return true;
        }

        for (int i = 0; i < t.length(); i++) {
            if (s.charAt(j) == t.charAt(i)) {
                j++;
            }
            if (j == s.length()) {
                return true;
            }
        }

        return false;
    }
}